R/dsl.R

```diff
@@ -51,13 +52,19 @@ uilabel <- function(..., type = "", is_link = TRUE) {
 #'      content = shiny::div("Second content"))
 #' ))
 tabset <- function(tabs,
+                   active = NULL,
                    id = generate_random_id("menu"),
                    menu_class = "top attached tabular",
                    tab_content_class = "bottom attached segment") {
-  identifiers <- replicate(length(tabs),
-                           list(id = generate_random_id("tab")),
-                           simplify = FALSE)
-  id_tabs <- purrr::map2(identifiers, tabs, ~ c(.x, .y))
+  id_tabs <- tabs %>% purrr::map(~ {
+    id = .x$id
+    menu = .x$menu
+    content = .x$content
+    list(id = if (!is.null(id)) id else tolower(gsub(" ", "_", menu)),
```

Comment:
Let's write separate function for setting up id:
function <- set_tab_id(tab) {
  id = .x$id
    menu = .x$menu
    content = .x$content
    list(
      id = if (!is.null(id)) id else tolower(gsub(" ", "_", menu)), 
      menu = menu, 
      content = content)
}

Furthermore there is one case not included here:
When menu is not string. We can setup menu as div element (check ??tabset). Let's use transformed menu as id only when it is character.

Reply:
done